src/components/labware/PacbioRunWell.vue

```diff
@@ -22,126 +22,190 @@
-        // Incomplete
-        return 'bg-failure text-white'
-      }
-      // Empty
-      return 'bg-gray-100 text-black'
-    },
+  plateNumber: {
+    type: Number,
+    required: true,
   },
-  methods: {
-    ...mapActions(usePacbioRunCreateStore, ['updateWell']),
-    onClick() {
-      this.$emit('click', this.position, this.plateNumber)
-    },
-    async drop(event) {
-      this.hover = false
-      await this.updatePoolLibraryBarcode(event.dataTransfer.getData('barcode'))
-    },
-    async updatePoolLibraryBarcode(barcode) {
-      const well = await this.getOrCreateWell(this.position, this.plateNumber)
-      const { id, type } = this.tubeContentByBarcode(barcode)
-      type === 'libraries' ? well.libraries.push(id) : type === 'pools' ? well.pools.push(id) : null
-      this.updateWell({ well: well, plateNumber: this.plateNumber })
-    },
+  interactive: {
+    type: Boolean,
+    required: true,
+    default: true,
   },
+})
+
+/*
+ * Define the emits for the component.
+ * The component emits a 'click' event when the well is clicked.
+ */
+const emit = defineEmits(['click'])
+
+/*
+ * Create a store instance of the pacbioRunCreateStore.
+ */
+const store = usePacbioRunCreateStore()
+
+/*
+ * Define refs for the component.
+ * The `hover` ref is used to determine if the well is being hovered over.
+ */
+const hover = ref(false)
+
+/*
+ * Computed property that returns the class names for the well.
+ * @returns {Array} - An array of class names for the well.
+ */
+const wellClassNames = computed(() => {
+  return [
+    status.value,
+    hover.value && props.interactive
+      ? 'ring ring-pink-600 ring-offset-1'
+      : 'border border-gray-800',
+    props.interactive ? 'cursor-pointer' : '',
+    'flex flex-col justify-center mx-auto rounded-full text-xs font-semibold aspect-square w-full select-none transition duration-200 ease-out',
+  ]
+})
+
+/*
+ * Computed property that returns the required metadata fields for the well.
+ * @returns {Array} - An array of required metadata fields for the well.
+ */
+const required_metadata_fields = computed(() => {
+  if (store.smrtLinkVersion.name == 'v11') {
+    return [
+      'movie_time',
+      'on_plate_loading_concentration',
+      'binding_kit_box_barcode',
+      'generate_hifi',
+    ]
+  } else if (store.smrtLinkVersion.name == 'v12_revio') {
+    return [
+      'movie_acquisition_time',
+      'include_base_kinetics',
+      'library_concentration',
+      'polymerase_kit',
+      'pre_extension_time',
+    ]
+  }
+  return []
+})
+
+/*
+ * Computed property that returns the well from the store.
+ * @returns {Object} - The well from the store.
+ */
+const storeWell = computed(() => {
+  const value = store.getWell(props.plateNumber, props.position)
+  return value ? { ...value } : undefined
+})
+
+/*
+ * Computed property that returns the tooltip for the well.
+ * @returns {string} - The tooltip for the well.
+ */
+const tooltip = computed(() => {
```

Comment:
I think I wrote this but it looks like it can be refactored. `[...storeWell.value.pools, ...storeWell.value.libraries]?.map(...).filter(Boolean).join(',') `. This should prevent the need to concat and map again.

Reply:
Good optimization 👍